behave/model_core.py

```diff
@@ -299,9 +304,28 @@ def reset(self):
         self.exc_traceback = None
         self.error_message = None
 
+    def send_status(self):
+        """Emit the volatile attributes of this model in a primitive dict
+        """
+        ret = {'exception': self.exception,
+               'error_message': self.error_message,
+               'exc_traceback': self.exc_traceback,
+               'captured': self.captured.send_status()
+               }
+        return ret
+
+    def recv_status(self, value):
+        """Set volatile attributes from a `send_status()` primitive value
+        """
+        for key in 'exception', 'error_message', 'exc_traceback':
+            if key in value:
+                setattr(self, key, value[key])
+        if 'captured' in value:
+            self.captured.recv_status(value['captured'])
+
     def store_exception_context(self, exception):
         self.exception = exception
-        self.exc_traceback = sys.exc_info()[2]
```

Comment:
is it a necessary change? It broke allure reporter(

Reply:
Necessary, as explained above. The tb object would not be serializable across processes. Might also benefit performance, because it frees any other objects bound to the traceback earlier on.